torch/_subclasses/fake_tensor.py

```diff
@@ -44,16 +44,22 @@ def from_tensor(t):
         existing_device = t.device
         return FakeTensor(t.to(device="meta"), existing_device)
 
-    @property
-    def device(self):
-        return self.fake_device
+    # TODO: resolve error in default __repr__
```

Comment:
What's the error?

Reply:
Haven't really looked into it yet, but this is the callstack and the `func` that is invoked:
```
aten.add.Tensor
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
prim.device.default
aten._reshape_alias.default
prim.device.default
prim.device.default
prim.device.default
aten.abs.default
prim.device.default
prim.device.default
prim.device.default
F
======================================================================
FAIL: test_zero_dim (__main__.FakeTensorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch/eellison/pytorch/test/test_fake_tensor.py", line 31, in test_zero_dim
    print(x)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_tensor.py", line 338, in __repr__
    return torch._tensor_str._str(self, tensor_contents=tensor_contents)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_tensor_str.py", line 481, in _str
    return _str_intern(self, tensor_contents=tensor_contents)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_tensor_str.py", line 447, in _str_intern
    tensor_str = _tensor_str(self, indent)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_tensor_str.py", line 270, in _tensor_str
    formatter = _Formatter(get_summarized_data(self) if summarize else self)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_tensor_str.py", line 103, in __init__
    nonzero_finite_vals = torch.masked_select(tensor_view, torch.isfinite(tensor_view) & tensor_view.ne(0))
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_subclasses/fake_tensor.py", line 98, in __torch_dispatch__
    return tree_map(partial(wrap, device=common_device), r)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/utils/_pytree.py", line 179, in tree_map
    return tree_unflatten([fn(i) for i in flat_args], spec)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/utils/_pytree.py", line 179, in <listcomp>
    return tree_unflatten([fn(i) for i in flat_args], spec)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_subclasses/fake_tensor.py", line 72, in wrap
    return FakeTensor(e, device)
  File "/private/home/eellison/anaconda3/lib/python3.8/site-packages/torch/_subclasses/fake_tensor.py", line 39, in __init__
    assert elem.device.type == "meta"
AssertionError
```

Reply:
Oh you need to add a special case for fake tensor in printing the same way we have a short circuit for meta printing. The meta printing short circuit no longer works because your device isn't meta anymore